File: npbloom-web/src/ui/components/meta/generateSentence.ts
// Moderately-long example words and names to make sure the click areas are large enough for untrained users
const wordChoices = [
  ['Alexis', 'Maurice', 'Nathan', 'Taylor'],
  ['likes', 'loves'],
  ['apples', 'cabbage', 'carrots', 'parsley']
];

const pickRandom = <T>(array: T[]): T => array[Math.floor(Math.random() * array.length)];

/**
 * Returns an example sentence to build a first tree out of.
 */
const generateSentence = () => wordChoices.map(pickRandom).join(' ');

export default generateSentence;

File: npbloom-core/src/commonMain/kotlin/ui/GenerateSentence.kt
@file:OptIn(ExperimentalJsExport::class)

package ui

import kotlin.js.ExperimentalJsExport
import kotlin.js.JsExport

// Moderately-long example words and names to make sure the click areas are large enough for untrained users
private val wordChoices = listOf(
    setOf("Alexis", "Maurice", "Nathan", "Taylor"),
    setOf("likes", "loves"),
    setOf("apples", "cabbage", "carrots", "parsley"),
)

/**
 * Returns an example sentence to build a first tree out of.
 */
@JsExport
fun generateSentence() = wordChoices.joinToString(" ") { it.random() }
